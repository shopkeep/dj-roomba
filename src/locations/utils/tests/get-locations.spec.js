const getLocations = require("../get-locations");

describe("get-locations", function() {
  beforeEach(function() {
    process.env.LOCATION_1_NAME = "New York";
    process.env.LOCATION_1_TIMEZONE = "America/New_York";
    process.env.LOCATION_1_SONOS_ADDRESS = "127.0.0.1";
    process.env.LOCATION_1_SPOTIFY_REGION = "US";
    process.env.LOCATION_1_SPOTIFY_MARKET = "US";
    process.env.LOCATION_1_DEFAULT_VOLUME = "42";

    process.env.LOCATION_2_NAME = "London";
    process.env.LOCATION_2_TIMEZONE = "Europe/London";
    process.env.LOCATION_2_SONOS_ADDRESS = "127.0.0.1";
    process.env.LOCATION_2_SPOTIFY_REGION = "EU";
    process.env.LOCATION_2_SPOTIFY_MARKET = "GB";
    process.env.LOCATION_2_DEFAULT_VOLUME = "50";
  });

  afterEach(function() {
    delete process.env.LOCATION_1_NAME;
    delete process.env.LOCATION_1_TIMEZONE;
    delete process.env.LOCATION_1_SONOS_ADDRESS;
    delete process.env.LOCATION_1_SPOTIFY_REGION;
    delete process.env.LOCATION_1_SPOTIFY_MARKET;
    delete process.env.LOCATION_1_DEFAULT_VOLUME;

    delete process.env.LOCATION_2_NAME;
    delete process.env.LOCATION_2_TIMEZONE;
    delete process.env.LOCATION_2_SONOS_ADDRESS;
    delete process.env.LOCATION_2_SPOTIFY_REGION;
    delete process.env.LOCATION_2_SPOTIFY_MARKET;
    delete process.env.LOCATION_2_DEFAULT_VOLUME;
  });

  it("generates a locations list from the environment variables", function() {
    expect(getLocations()).toMatchSnapshot();
  });
});
