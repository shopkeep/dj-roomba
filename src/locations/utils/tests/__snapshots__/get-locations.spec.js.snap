// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-locations generates a locations list from the environment variables 1`] = `
Array [
  Object {
    "defaultVolume": "42",
    "device": Sonos {
      "_events": Object {
        "newListener": [Function],
      },
      "_eventsCount": 1,
      "host": "127.0.0.1",
      "options": Object {
        "endpoints": Object {
          "device": "/DeviceProperties/Control",
          "rendering": "/MediaRenderer/RenderingControl/Control",
          "transport": "/MediaRenderer/AVTransport/Control",
        },
        "spotify": Object {
          "region": "3079",
        },
      },
      "port": 1400,
    },
    "name": "New York",
    "sonosAddress": "127.0.0.1",
    "spotifyMarket": "US",
    "spotifyRegion": "US",
    "timezone": "America/New_York",
  },
  Object {
    "defaultVolume": "50",
    "device": Sonos {
      "_events": Object {
        "newListener": [Function],
      },
      "_eventsCount": 1,
      "host": "127.0.0.1",
      "options": Object {
        "endpoints": Object {
          "device": "/DeviceProperties/Control",
          "rendering": "/MediaRenderer/RenderingControl/Control",
          "transport": "/MediaRenderer/AVTransport/Control",
        },
        "spotify": Object {
          "region": "2311",
        },
      },
      "port": 1400,
    },
    "name": "London",
    "sonosAddress": "127.0.0.1",
    "spotifyMarket": "GB",
    "spotifyRegion": "EU",
    "timezone": "Europe/London",
  },
]
`;
